### DISCUSSION TITLE
Tidyselect targets for tar_combine

### DISCUSSION THREAD

**Discussion by riazarbi** [ropensci/tarchetypes#35] [Help] [answered]
Hello,

I pepper my targets list with tests, which basically check that a particular target passes certain tests. 

Here's an example -

```
  tar_target(
    lis_spatial_unique.test,
    test_df(lis_spatial_unique, lis_key, error = "stop")),
```

Here's a result -

```
> tar_read(lis_spatial_unique.test)
              object is_dataframe has_unique_column overall_test
1 lis_spatial_unique         TRUE              TRUE         TRUE
```

Now, I have many of these tests, which are all named according to the pattern `*.test`:

```
> tar_meta(names = ends_with(".test"), fields = name)
# A tibble: 7 x 1
  name                                     
  <chr>                                    
1 property.residential_income_estimate.test
2 rate_categories.test                     
3 lis_spatial_unique.test                  
4 contracts_tidy.test                      
5 property.minimal.test                    
6 contracts_ewkt.test                      
7 property.test  
```

I would like to combine all these result dataframes into a combined dataframe.

```
  tar_target(test_names, 
             tar_meta(names = ends_with(".test"), fields = name) %>% pull(name) %>% as.list(),
             format = "rds"),
  tar_combine(test_results,
              test_names)

```

All of these targets have the same structure. I would like to add more test targets and not think about adding them to the test_results target

Unfortunately, I'm having trouble passing the list of test targets to the `test_results` target. 

Is there a way to do this that I'm missing?

**Comment by wlandau** [comment, marked as answer]
Target objects generated by [`tar_map()`](https://books.ropensci.org/targets/static.html#map) have names, so you could pre-filter the output of `tar_map()` and feed that to `tar_combine()`. 

Or, if you are not using `tar_map()`, you could write your testing targets in a separate list and feed that to `tar_combine()`.

```r
# _targets.R
library(targets)
library(tarchetypes)

other_targets <- list(
  tar_target(x, ...)
  tar_target(y,...)
  ...
)

testing_targets <- list(
  tar_target(a.test, ...)
  tar_target(b.test, ...)
)

testing_results <- tar_combine(results, testing_targets)

# Full list of targets:
list(
  other_targets,
  testing_targets,
  testing_results
)
```

Another option is to use [dynamic branching](https://books.ropensci.org/targets/dynamic.html) because it automatically combines data frame return values.

I would advise against using `tar_meta()` inside a target's command.

**Comment by riazarbi** [comment]
Thanks, the separate lists works well.
